refresh/repo.go

```diff
@@ -41,11 +43,11 @@ func DefaultRefreshTokenGenerator() ([]byte, error) {
 type RefreshTokenRepo interface {
 	// Create generates and returns a new refresh token for the given client-user pair.
 	// On success the token will be return.
-	Create(userID, clientID string) (string, error)
+	Create(userID, clientID string, scope []string) (string, error)
 
-	// Verify verifies that a token belongs to the client, and returns the corresponding user ID.
+	// Verify verifies that a token belongs to the client and is valid for the given scopes. It returns the user ID to which the token belongs.
 	// Note that this assumes the client validation is currently done in the application layer,
-	Verify(clientID, token string) (string, error)
+	Verify(clientID, token string, scopes []string) (string, error)
```

Comment:
Ah never mind. I didn't realize that scopes could change between refresh requests.

Can we make this comment more explicit about what it's doing? I just want to make sure someone working with this can understand exactly where they are in the refresh token flow. E.g.

```
Verify parses a raw refresh token and returns the user ID for which the token was issued for. It also ensures the requested scopes do not include any scopes not in the original request.
NOTE: Refresh requests require validating the client_id and client_secret pair, which callers MUST do before calling this function.
```